Notification.Application/Commands/Broker/SubscriptionCommand.cs

```diff
@@ -0,0 +1,75 @@
+﻿using MediatR;
+using Notification.Application.Handler;
+using Notification.Application.Logger;
+using Notification.Contracts.DataTransferObject.Broker;
+using Notification.Data.Entities;
+using Notification.Data.Repositories.Interface;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Notification.Application.Commands.Broker
+{
+    public class SubscriptionCommand : IRequest
+    {
+        public SubscriptionCommand(Subscription subscription)
+        {
+            Subscription = subscription;
+        }
+
+        public Subscription Subscription { get; set; }
+    }
+
+    public class SubscriptionCommandHandler : LoggerRequestHandler<SubscriptionCommand, Unit>
+    {
+        private readonly IBookRepository _bookRepository;
+        private readonly IUserRepository _userRepository;
+
+        public SubscriptionCommandHandler(ILoggerManager logger, IBookRepository bookRepository, IUserRepository userRepository)
+            : base(logger)
+        {
+            _bookRepository = bookRepository;
+            _userRepository = userRepository;
+        }
+
+        public async override Task<Unit> HandleInternalAsync(SubscriptionCommand request, CancellationToken ct)
+        {
+            var entityUser = await _userRepository.GetOneByCondition(x => x.Id == request.Subscription.UserId, ct);
+            var entityBook = await _bookRepository.GetOneByCondition(x => x.Id == request.Subscription.BookId, ct);
+
+            if (entityUser is null)
```

Comment:
put it into separate method, like GetUser

1) return if exitsts await _userRepository.GetOneByCondition(x => x.Id == request.Subscription.UserId, ct);
2) if not create new entity/add to db and return from acition

Reply:
but it's up to you. You can leave as it is.